### ISSUE TITLE
Non existent extra damaged ship(s) detected

### ISSUE THREAD

**Issue by CripsyBekon** [perryhuynh/kcauto#272]
#### Environment details

* kcauto version (version on first line of `CHANGELOG.md`):6.6.1
* Viewer/browser:Chromium
* Operating system and/or VM:Linux

#### Issue variability
Is the issue consistent (can be reproduced consistently) or is it intermittent (only happens sometimes)?:
Consistent

#### Issue

At this point i am not sure what is actually wrong with it. I took a look back and i found that the damage reporting is having issues which leads to the repair function going on an infinite loop while trying to find a minor damaged ship that dint exist.

![Log 1 Node M](https://user-images.githubusercontent.com/32804784/45594745-1b356900-b9e3-11e8-909c-0fe61e4a98db.png)

Notice that in this picture, i have 1 moderate damaged ship and 2 minor damaged ship. on L28 of the log, it only stated 1 moderate and 1 minor.

![Log 1 Node O](https://user-images.githubusercontent.com/32804784/45594598-fb9d4100-b9e0-11e8-8487-d190b2d0604e.png)

Now on Node O, i have 5 heavy damaged ships and 1 moderate damage ships. L38 states that i have 5 heavy, 1 moderate and 1 light.

![Log 2 Node F](https://user-images.githubusercontent.com/32804784/45594687-3489e580-b9e2-11e8-86db-e25a4859e26d.png)

A second attempt was initiated in order to reproduce the problem. In the picture i only have 1 moderate damaged ship. On the second log provided, it stated that i have 1 moderate and 1 light damaged ship.
On the third node i have 2 moderate and 1 light damaged ship, which matches the amount of damaged ship. 

![Log 2 Node O](https://user-images.githubusercontent.com/32804784/45594730-c691ee00-b9e2-11e8-8620-393667d6f347.png)

On Node O i ended up with 2 heavy and 3 moderate damaged ships. But somehow 3 minor ships managed to get registered.

#### Log
**Log 1**
```
[2018-09-16 18:03:19] Beginning combat sortie.
[2018-09-16 18:03:24] Assign LBAS groups to nodes.
[2018-09-16 18:03:24] Assigning nodes for LBAS group 1.
[log] CLICK on L(1003,171)@S(0)[0,0 1920x1022] (526 msec)
[2018-09-16 18:03:39] Fleet at Node #1
[2018-09-16 18:03:39] Custom formation specified for node #1.
[2018-09-16 18:03:39] Engaging the enemy in vanguard formation.
[log] CLICK on L(1022,611)@S(0)[0,0 1920x1022] (525 msec)
[2018-09-16 18:04:36] Fleet 1 damage counts: 0 heavy / 0 moderate / 0 minor
[2018-09-16 18:04:48] Continuing sortie.
[log] CLICK on L(473,481)@S(0)[0,0 1920x1022] (521 msec)
[2018-09-16 18:05:07] Fleet at node F.
[2018-09-16 18:05:07] Selecting Node E from Node F.
[2018-09-16 18:05:17] Fleet at Node #2
[2018-09-16 18:05:17] Custom formation specified for node #2.
[2018-09-16 18:05:17] Engaging the enemy in diamond formation.
[log] CLICK on L(1137,364)@S(0)[0,0 1920x1022] (523 msec)
[2018-09-16 18:05:52] Fleet 1 damage counts: 0 heavy / 0 moderate / 1 minor
[2018-09-16 18:06:03] Continuing sortie.
[log] CLICK on L(419,412)@S(0)[0,0 1920x1022] (523 msec)
[2018-09-16 18:06:32] Fleet at Node #3
[2018-09-16 18:06:32] Custom formation specified for node #3.
[2018-09-16 18:06:32] Engaging the enemy in vanguard formation.
[log] CLICK on L(1069,598)@S(0)[0,0 1920x1022] (523 msec)
[2018-09-16 18:07:54] No night battle specified for node #3.
[2018-09-16 18:07:54] Declining night battle.
[log] CLICK on L(487,426)@S(0)[0,0 1920x1022] (521 msec)
[2018-09-16 18:08:08] Fleet 1 damage counts: 0 heavy / 1 moderate / 1 minor
[2018-09-16 18:08:19] Continuing sortie.
[log] CLICK on L(418,476)@S(0)[0,0 1920x1022] (524 msec)
[2018-09-16 18:08:47] Fleet at Node #4
[2018-09-16 18:08:47] Custom formation specified for node #4.
[2018-09-16 18:08:47] Engaging the enemy in line ahead formation.
[log] CLICK on L(716,374)@S(0)[0,0 1920x1022] (523 msec)
[2018-09-16 18:10:48] Custom night battle specified for node #4.
[2018-09-16 18:10:48] Commencing night battle.
[log] CLICK on L(786,450)@S(0)[0,0 1920x1022] (521 msec)
[2018-09-16 18:11:22] Fleet 1 damage counts: 5 heavy / 1 moderate / 1 minor
[2018-09-16 18:11:34] Sortie complete. Encountered 4 combat nodes (nodes 1, 2, 3, 4).
[2018-09-16 18:11:35] An expedition fleet has returned. Probably fleet 3
[2018-09-16 18:11:51] At home
[2018-09-16 18:11:51] Navigating to repair screen.
--standard repair before loop--
[2018-09-16 18:12:55] Picking damaged ship from combat fleet(s) to repair.
[log] CLICK on L(486,568)@S(0)[0,0 1920x1022] (525 msec)
[2018-09-16 18:13:04] Picking damaged ship from combat fleet(s) to repair.
[log] CLICK on L(457,451)@S(0)[0,0 1920x1022] (521 msec)
[2018-09-16 18:13:14] Picking damaged ship from combat fleet(s) to repair.
[log] CLICK on L(379,561)@S(0)[0,0 1920x1022] (522 msec)
[2018-09-16 18:13:23] Picking damaged ship from combat fleet(s) to repair.
[log] CLICK on L(305,447)@S(0)[0,0 1920x1022] (524 msec)
[2018-09-16 18:13:34] Picking damaged ship from combat fleet(s) to repair.
[log] CLICK on L(357,594)@S(0)[0,0 1920x1022] (521 msec)
[2018-09-16 18:13:43] Picking damaged ship from combat fleet(s) to repair.
[log] CLICK on L(458,438)@S(0)[0,0 1920x1022] (522 msec)
[2018-09-16 18:13:53] Picking damaged ship from combat fleet(s) to repair.
[log] CLICK on L(488,567)@S(0)[0,0 1920x1022] (522 msec)
[2018-09-16 18:14:02] Picking damaged ship from combat fleet(s) to repair.
[KAGA] [2018-09-16 18:14:06] [c.w.k.k.KCAuto] INFO - Terminating current KCAuto session
[KAGA] [2018-09-16 18:14:06] [c.w.k.k.KCAuto] INFO - KCAuto session has terminated!
```
**Log 2**
```
[2018-09-16 18:23:50] Engaging the enemy in diamond formation.
[log] CLICK on L(1085,350)@S(0)[0,0 1920x1022] (522 msec)
[2018-09-16 18:24:30] Fleet 1 damage counts: 0 heavy / 1 moderate / 1 minor
[2018-09-16 18:24:41] Continuing sortie.
[log] CLICK on L(465,431)@S(0)[0,0 1920x1022] (525 msec)
[2018-09-16 18:25:09] Fleet at Node #3
[2018-09-16 18:25:09] Custom formation specified for node #3.
[2018-09-16 18:25:09] Engaging the enemy in vanguard formation.
[log] CLICK on L(1025,609)@S(0)[0,0 1920x1022] (521 msec)
[2018-09-16 18:26:41] No night battle specified for node #3.
[2018-09-16 18:26:41] Declining night battle.
[log] CLICK on L(460,425)@S(0)[0,0 1920x1022] (525 msec)
[2018-09-16 18:26:54] Fleet 1 damage counts: 0 heavy / 2 moderate / 1 minor
[2018-09-16 18:27:05] Continuing sortie.
[log] CLICK on L(484,446)@S(0)[0,0 1920x1022] (523 msec)
[2018-09-16 18:27:35] Fleet at Node #4
[2018-09-16 18:27:35] Custom formation specified for node #4.
[2018-09-16 18:27:35] Engaging the enemy in line ahead formation.
[log] CLICK on L(630,370)@S(0)[0,0 1920x1022] (521 msec)
^[[1;5C[2018-09-16 18:29:10] Custom night battle specified for node #4.
[2018-09-16 18:29:10] Commencing night battle.
[log] CLICK on L(808,448)@S(0)[0,0 1920x1022] (526 msec)
[2018-09-16 18:29:50] Fleet 1 damage counts: 2 heavy / 3 moderate / 3 minor
```
#### Config
```
RepairLimit = minor
```

#### Full Log Files if you are interested
[Log1.txt](https://github.com/mrmin123/kcauto/files/2386433/Log1.txt)
[Log2.txt](https://github.com/mrmin123/kcauto/files/2386434/Log2.txt)

Really sorry about the inconsistent crop size, i have to prepare my dinner.

**Comment by mrmin123** [comment]
Thanks for the report. This is actually related to/the root cause of #264. I'm heading to bed but will take a look tomorrow afternoon.

**Comment by CripsyBekon** [comment]
weird. I dint get it on 6.6.0 but i could on 6.6.1. thanks anyways mrmin! have a good rest

**Comment by mrmin123** [comment]
@CripsyBekon could you try raising the `DAMAGE_SIMILARITY` to `0.88` in `kca_globals.py`?

**Comment by CripsyBekon** [comment]
sure i will get back to you in about an hour.

**Comment by CripsyBekon** [comment]
sorry for the wait i tested a few times to make sure but it works now after changing `DAMAGE_SIMILARITY` to `0.88`

**Comment by mrmin123** [comment]
Thanks, resolved in 6.6.2.
